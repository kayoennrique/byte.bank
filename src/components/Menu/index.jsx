import React from 'react';
import styles from './Menu.module.css';

const listMenu = [
  'Inicial',
  'Transferências',
  'Investments',
  'Outros serviços',
];

export default function Menu() {
  return (
    <nav className={styles.menu}>
      {listMenu.map((item, index) => {
        return (
          <div key={item} className={styles.item}>
            <a href="/" className={styles.link}>
              {item}
            </a>
            {index !== listMenu.length - 1 && (
              <div className={styles.divisor} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
