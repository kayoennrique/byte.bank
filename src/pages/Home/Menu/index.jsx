import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Menu.module.css';

const listMenu = [
  { link: 'Início', href: '/home' },
  { link: 'Minha conta', href: '/my-account' },
  { link: 'Cartões', href: '/home/cards' },
  { link: 'Serviços', href: '/home/services' },
  { link: 'Investimentos', href: '/home/investiments' },
];

export default function Menu({ path }) {
  return (
    <nav className={styles.menu}>
      {listMenu.map((item, index) => {
        return (
          <div key={item.href} className={styles.item}>
            <Link
              to={item.href}
              className={`${styles.link} ${
                path === item.href && styles.linkActive
              }`}
            >
              {item.link}
            </Link>
            {index !== listMenu.length - 1 && (
              <div className={styles.divisor} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
