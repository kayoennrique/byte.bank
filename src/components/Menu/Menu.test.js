import { render, screen } from '@testing-library/react';
import Menu from './index';

test('Should render a link to the home page', () => {
  render(<Menu />);
  const linkHomePage = screen.getByText('Inicial');
  expect(linkHomePage).toBeInTheDocument();
});

test('Should render a list of links', () => {
  render(<Menu />);
  const linksList = screen.getAllByRole('link');
  expect(linksList).toHaveLength(4);
});

test('Should not render the link to Extract', () => {
  render(<Menu />);
  const linkExtract = screen.queryByText('Extrato');
  expect(linkExtract).not.toBeInTheDocument();
});

test('It should render a list of links and with the link class', () => {
    render(<Menu />);
    const links = screen.getAllByRole('link');
    links.forEach((link) => expect(link).toHaveClass('link'));
    expect(links).toMatchSnapshot();
});