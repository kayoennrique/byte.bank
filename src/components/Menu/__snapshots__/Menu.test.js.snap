// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render a list of links and with the link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investments
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
